### PR TITLE
chore: Upgrades to .NET 6

### PR DESCRIPTION
**Pull request by georg-eckert-zeiss**
Makes MeshModel compatible with .NET 6 by fixing a [breaking change](https://docs.microsoft.com/en-us/dotnet/core/compatibility/core-libraries/6.0/partial-byte-reads-in-streams).

Adds more tests.

### CODE REVIEW

File: src/MeshModel/ReaderWriterExtensions.cs [hunk 1]
```diff
@@ -411,7 +411,14 @@ private static IEnumerable<(int Count, int Current, byte[] buffer)> ReadByteArra
 				if( count <= 0 )
 					break;
 
-				rdr.BaseStream.Read( buffer, 0, count );
+				var totalRead = 0;
```
Comment (czdietrich) [inline src/MeshModel/ReaderWriterExtensions.cs:414]: Formatting seams not yet correct.

File: src/Tests/MeshModel.Tests.csproj [hunk 2]
```diff
@@ -34,6 +34,14 @@
 
     <ItemGroup>
         <Content CopyToOutputDirectory="PreserveNewest" Include="TestData\*.meshModel" />
+        <Content CopyToOutputDirectory="PreserveNewest" Include="TestData\Edges.dat" />
```
Comment (czdietrich) [inline src/Tests/MeshModel.Tests.csproj:37]: Maybe we just reduce this to `<Content CopyToOutputDirectory="PreserveNewest" Include="TestData\*" />`?

File: src/Tests/MeshTest.cs [hunk 3]
```diff
@@ -172,50 +311,60 @@ public void ImportExportTest()
...
-					for( var i = 0; i < originalBuffer.Length; ++i )
+					var commonReadLength = Math.Min( readByteCountOriginal, readByteCountReWritten );
+					for( var i = 0; i < count; ++i )
 					{
 						byteWiseEqual &= originalBuffer[ i ] == reWrittenBuffer[ i ];
 						if( !byteWiseEqual )
 							break;
 					}
+
+					if( !byteWiseEqual )
```
Comment (czdietrich) [inline src/Tests/MeshTest.cs:364]: Do we actually need this check?

File: src/Tests/AssertingStream.cs [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+#endregion
+
+namespace Zeiss.PiWeb.MeshModel.Tests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using NUnit.Framework;
+    public sealed class AssertingStream : Stream
```
Comment (czdietrich) [inline src/Tests/AssertingStream.cs:18]: Maybe you could add an XML-comment for clarification.

**Review by czdietrich — APPROVED** [review]

**Comment by czdietrich** [comment]
I would state `fix:` instead of `chore:` since reading from stream was actually wrong. The interface was from the beginning designed the way to not make assumptions about the amount of read bytes for a single `Read`-invocation.

File: src/MeshModel/ReaderWriterExtensions.cs (hunk 1 above)
Comment (georg-eckert-zeiss) [reply src/MeshModel/ReaderWriterExtensions.cs:414]: Something seems not right with our DotSettings. I'll look into it.

Comment (georg-eckert-zeiss) [reply src/MeshModel/ReaderWriterExtensions.cs:414]: I'll repair this in a separate PR.

File: src/Tests/MeshTest.cs (hunk 3 above)
Comment (georg-eckert-zeiss) [reply src/Tests/MeshTest.cs:364]: Not if we are using the `AssertingStream`. I'll change that.

Comment (georg-eckert-zeiss) [reply src/Tests/MeshTest.cs:364]: But you are right. This check is not necessary, since it is always false :D

**Pull request merged**